python/plugins/processing/gui/BatchAlgorithmDialog.py

```diff
@@ -57,9 +57,17 @@ def __init__(self, alg, parent=None):
     def runAsSingle(self):
         self.close()
 
-        from processing.gui.AlgorithmDialog import AlgorithmDialog
+        if self.algorithm().id().startswith("gdal"):
+            from processing.algs.gdal.GdalAlgorithmDialog import GdalAlgorithmDialog
+
+            dlg = GdalAlgorithmDialog(
+                self.algorithm().create(), parent=iface.mainWindow()
+            )
```

Comment:
Instead of hard-coding gdal specific logic here, can you use self.algorithm().createCustomParametersWidget() instead?